python_modules/dagster/dagster/_core/definitions/cacheable_assets.py

```diff
@@ -348,26 +346,10 @@ def __init__(
         self._resource_defs = resource_defs
 
         super().__init__(
-            unique_id=f"{wrapped.unique_id}_resources_{self._get_hash()}",
+            unique_id=wrapped.unique_id,
```

Comment:
> I can't think of a situation where a CacheableAssetsDefinition would ever need to co-exist with itself with with_resources

That is certainly true, but we can at the very least make the unique id `{wrapped.unique_id}_with_resources` just to sidestep that entirely. But yes there's really no way to have identical cacheable assets that only differ in resource config (because then their asset keys would be the same, so there'd be a conflict)

Reply:
good point, updated.

Reply:
good call, updated